src/libs/auth.ts

```diff
@@ -55,16 +55,62 @@ const getAuthInstance = () => {
   return authStore.get().authContext;
 };
 
-export const signOut = () => {
-  // TODO: invalidate runtime cookies https://broadworkbench.atlassian.net/browse/IA-3498
+export const enum SignOutCauses {
+  requested,
+  disabled,
+  declinedTos,
+  expiredRefreshToken,
+  errorRefreshingAuthToken,
+  idleStatusMonitor,
+  unspecified,
+}
+
+const sendSignOutMetrics = (causes: SignOutCauses): void => {
+  const eventToFire = (() => {
+    switch (causes) {
+      case SignOutCauses.requested: {
+        return Events.user.signOut.requested;
+      }
+      case SignOutCauses.disabled: {
+        return Events.user.signOut.disabled;
+      }
+      case SignOutCauses.declinedTos: {
+        return Events.user.signOut.declinedTos;
+      }
+      case SignOutCauses.expiredRefreshToken: {
+        return Events.user.signOut.expiredRefreshToken;
+      }
+      case SignOutCauses.errorRefreshingAuthToken: {
+        return Events.user.signOut.errorRefreshingAuthToken;
+      }
+      case SignOutCauses.unspecified: {
+        return Events.user.signOut.unspecified;
+      }
+      default: {
+        return Events.user.signOut.unspecified;
+      }
+    }
+  })();
   const sessionEndTime: number = Date.now();
-  const tokenMetadata = authStore.get().authTokenMetadata;
-  Ajax().Metrics.captureEvent(Events.userLogout, {
+  const authStoreState = authStore.get();
+  const tokenMetadata = authStoreState.authTokenMetadata;
+
+  Ajax().Metrics.captureEvent(eventToFire, {
     sessionEndTime: Utils.makeCompleteDate(sessionEndTime),
-    sessionDurationInSeconds: (sessionEndTime - authStore.get().sessionStartTime) / 1000.0,
+    sessionDurationInSeconds: (sessionEndTime - authStoreState.sessionStartTime) / 1000.0,
     authTokenCreatedAt: Utils.formatTimestampInSeconds(tokenMetadata.createdAt),
     authTokenExpiresAt: Utils.formatTimestampInSeconds(tokenMetadata.expiresAt),
+    totalAuthTokensUsedThisSession: authStoreState.authTokenMetadata.totalAuthTokensUsedThisSession,
+    totalAuthTokenLoadAttemptsThisSession: authStoreState.authTokenMetadata.totalAuthTokenLoadAttemptsThisSession,
   });
+};
+
+export const signOut = (causes: SignOutCauses = SignOutCauses.unspecified): void => {
```

Comment:
```suggestion
export const signOut = (cause: SignOutCauses = SignOutCauses.unspecified): void => {
```

Similar to previous comments, there's only one cause here.